src/external.js

```diff
@@ -47,8 +47,28 @@ External.prototype.fetchOrCreate = function () {
   return this._metadata.fetch().then(createOrPopulate.bind(this)).catch(fetchFailed.bind(this));
 };
 
+External.prototype.fetch = function () {
+  var Populate = function (object) {
+    this.success = true;
+    if (object !== null) {
+      this._coinify = object.coinify ? new Coinify(object.coinify, this, this._delegate) : undefined;
+    }
+    return this;
+  };
+  var fetchFailed = function (e) {
+    // Metadata service is down or unreachable.
+    this.success = false;
+    return Promise.reject(e);
```

Comment:
@pernas this causes wallet login to fail if metadata is down. `fetchOrCreate()` didn't reject. I changed MyWallet to catch this and proceed with login in 7955ef94d73.